#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: upstream patch bash30-008

			     BASH PATCH REPORT
			     =================

Bash-Release: 3.0
Patch-ID: bash30-008

Bug-Reported-by: uberlord@rsm.demon.co.uk
Bug-Reference-ID: <1092327965.4233.1.camel@uberlaptop.ubernet>
Bug-Reference-URL: http://lists.gnu.org/archive/html/bug-bash/2004-08/msg00144.html

Bug-Description:

> Description:
>         Bash 3 breaks array expansion
> 
> Repeat-By:
>         #!/bin/bash
>         x=(one two)
>         echo ${x[@]:1}
>         # prints nothing in bash 3
>         # prints two in bash 2

Patch:

*** ../bash-3.0/subst.c	Sun Jul  4 13:56:13 2004
--- subst.c	Thu Aug 12 13:36:17 2004
***************
*** 4892,4896 ****
      *e1p += len;
  
!   if (*e1p >= len || *e1p < 0)
      return (-1);
  
--- 4912,4916 ----
      *e1p += len;
  
!   if (*e1p > len || *e1p < 0)
      return (-1);
  

*** ../bash-3.0/patchlevel.h	Wed Aug 22 08:05:39 2001
--- patchlevel.h	Thu Sep  2 15:04:32 2004
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 7
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 8
  
  #endif /* _PATCHLEVEL_H_ */
